### PR TITLE
14162_asset_manager_4_6_1

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
Added a SwrveAssetManager which is the same download asset functionality from SwrveImp, but now easier to test and swap out.

Also added tests to for this for which there was none.

https://swrvedev.jira.com/browse/SWRVE-14162

@Sergio-Mira @seaders @adam-govan

### CODE REVIEW

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java [hunk 1]
```diff
@@ -118,7 +119,7 @@ protected T init(final Activity activity) throws IllegalArgumentException {
             appStoreURLs = new SparseArray<String>();
 
             // Find cache folder
```
Comment (adam-govan) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: change comment here

**Comment by adam-govan** [comment]
🚀

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java (hunk 1 above)
Comment (dominicmarmionswrve) [reply SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: Will do!
